src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/GeneratedDocumentPublisher.cs

```diff
@@ -98,10 +98,11 @@ public void PublishCSharp(ProjectKey projectKey, string filePath, SourceText sou
         {
             HostDocumentFilePath = filePath,
             ProjectKeyId = projectKey.Id,
-            Changes = textChanges.Select(static t => t.ToRazorTextChange()).ToArray(),
+            Changes = [.. textChanges.Select(static t => t.ToRazorTextChange())],
             HostDocumentVersion = hostDocumentVersion,
+            PreviousHostDocumentVersion = previouslyPublishedData.HostDocumentVersion,
             PreviousWasEmpty = previouslyPublishedData.SourceText.Length == 0,
-            Checksum = Convert.ToBase64String(sourceText.GetChecksum().ToArray()),
+            Checksum = Convert.ToBase64String([.. sourceText.GetChecksum()]),
```

Comment:
I thought about adding checksums, then decided it would be too hard, then saw that Andrew had already added them and I forgotten, but then decided that waiting for the right version would be better, so we could wait for the provisional completion revert, rather than just rejecting a request because the checksum doesn't match.